src/main/java/com/example/springbootbookstore/model/Book.java

```diff
@@ -7,9 +7,20 @@
 import jakarta.persistence.Id;
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
-import java.util.Objects;
-
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.Filter;
+import org.hibernate.annotations.FilterDef;
+import org.hibernate.annotations.ParamDef;
+import org.hibernate.annotations.SQLDelete;
+
+@Getter
+@Setter
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@FilterDef(name = "deletedBookFilter",
+        parameters = @ParamDef(name = "isDeleted", type = Boolean.class))
+@Filter(name = "deletedBookFilter", condition = "is_deleted = :isDeleted")
```

Comment:
```suggestion
@SQLRestriction(value = "is_deleted=false")
```
fix in all places where needed

Reply:
fixed